tensornetwork/block_sparse/block_tensor_test.py

```diff
@@ -223,3 +223,80 @@ def test_find_diagonal_sparse_blocks(num_legs, num_charges):
 
   assert np.sum(np.prod(ss, axis=0)) == np.sum([len(b) for b in bs])
   np.testing.assert_allclose(unique_left, cs.charges)
+
+
+@pytest.mark.parametrize('num_legs', [2, 3, 4])
+@pytest.mark.parametrize('num_charges', [1, 2, 3])
+def test_find_transposed_diagonal_sparse_blocks(num_legs, num_charges):
+  np.random.seed(10)
+  order = np.arange(num_legs)
+  while np.all(order == np.arange(num_legs)):
+    np.random.shuffle(order)
```

Comment:
Why do you have this in a while loop?

Reply:
wanted to avoid  to make my own choice. I can just hardcode the result of the shuffle if you want.